features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/RecordingTimeBank.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder
+
+import java.util.concurrent.TimeUnit
+import kotlin.math.min
+
+/**
+ * Implementation of [TimeBank], it maintains a time balance which can be consumed for session replay recording.
+ * Every time the session replay recording task tries to execute, [updateAndCheck] is called to increase the time
+ * balance according to the time passed since last call, and check if the balance is positive to execute the task.
+ */
+internal class RecordingTimeBank(
+    private val maxTimeBalancePerSecondInMs: Long = DEFAULT_MAX_TIME_BALANCE_PER_SEC_IN_MS
+) : TimeBank {
+
+    @Volatile
+    private var recordingTimeBalanceInNano = TimeUnit.MILLISECONDS.toNanos(maxTimeBalancePerSecondInMs)
+
+    @Volatile
+    private var lastStartTime: Long = 0
+
+    override fun consume(executionTime: Long) {
+        recordingTimeBalanceInNano -= executionTime
+    }
+
+    override fun updateAndCheck(timestamp: Long): Boolean {
+        increaseTimeBank(timestamp)
+        lastStartTime = timestamp
+        return recordingTimeBalanceInNano >= 0
+    }
+
+    private fun increaseTimeBank(timestamp: Long) {
+        val timePassedSinceLastExecution = timestamp - lastStartTime
+        // The factor of balance increasing by time. if increasing 100ms balance in the bank takes 1000ms, then the
+        // factor will be 100ms/1000ms = 0.1f
+        val balanceFactor = maxTimeBalancePerSecondInMs.toFloat() / TimeUnit.SECONDS.toMillis(1)
```

Comment:
I am not sure which one is the variable here ? it seems that this has always the same value as `maxTimeBalancePerSecondInMs` will never change ? Am I wrong ?

Reply:
Indeed `balanceFactor` will never change, I add this variable as an intermediate argument to avoid a long formula in the code, trying to make it understandable, what do you think?

Reply:
In that case isn't more readable if you declare only once in your class and keep it as a `val` ?

Reply:
updated